File: src/main/java/org/chillout1778/subsystems/Drive.java
package org.chillout1778.subsystems;

import com.revrobotics.CANSparkMax;
import com.revrobotics.CANSparkMaxLowLevel;
import edu.wpi.first.wpilibj2.command.Command;
import edu.wpi.first.wpilibj2.command.SubsystemBase;
import org.chillout1778.Constants;
import org.chillout1778.commands.DriveCommand;

public class Drive extends SubsystemBase {
    private CANSparkMax leftParentMotor = new CANSparkMax(1, CANSparkMaxLowLevel.MotorType.kBrushless);
    private CANSparkMax leftChildMotor = new CANSparkMax(2, CANSparkMaxLowLevel.MotorType.kBrushless);

    private CANSparkMax rightParentMotor = new CANSparkMax(3, CANSparkMaxLowLevel.MotorType.kBrushless);
    private CANSparkMax rightChildMotor = new CANSparkMax(4, CANSparkMaxLowLevel.MotorType.kBrushless);

    public Drive(){
        setDefaultCommand(new DriveCommand(this));
        leftChildMotor.follow(leftParentMotor);
        rightChildMotor.follow(rightParentMotor);
    }

    public void setRightSpeed(double speed){
        rightParentMotor.set(speed);
    }
    public void setLeftSpeed(double speed) {
        leftParentMotor.set(speed);
    }

    public void setBothSpeeds(double speed){
        setLeftSpeed(speed);
        setRightSpeed(speed);
    }
}

File: src/main/java/org/chillout1778/subsystems/Drive.kt
package org.chillout1778.subsystems

import com.revrobotics.CANSparkMax
import com.revrobotics.CANSparkMaxLowLevel
import edu.wpi.first.wpilibj2.command.SubsystemBase
import org.chillout1778.commands.DriveCommand

object Drive : SubsystemBase() {
    private val leftParentMotor = CANSparkMax(1, CANSparkMaxLowLevel.MotorType.kBrushless)
    private val leftChildMotor = CANSparkMax(2, CANSparkMaxLowLevel.MotorType.kBrushless)
    private val rightParentMotor = CANSparkMax(3, CANSparkMaxLowLevel.MotorType.kBrushless)
    private val rightChildMotor = CANSparkMax(4, CANSparkMaxLowLevel.MotorType.kBrushless)

    init {
        defaultCommand = DriveCommand()
        leftChildMotor.follow(leftParentMotor)
        rightChildMotor.follow(rightParentMotor)
    }

    fun setRightSpeed(speed: Double) {
        rightParentMotor.set(speed)
    }

    fun setLeftSpeed(speed: Double) {
        leftParentMotor.set(speed)
    }

    fun setBothSpeeds(speed: Double){
        setRightSpeed(speed)
        setLeftSpeed(speed)
    }
}
